internal/cmd/cmd_test.go

```diff
@@ -212,6 +220,72 @@ func TestCmdStdinBlocked(t *testing.T) {
 	}
 }
 
+func TestCmdAfterExitFun(t *testing.T) {
+	cmd := Command(&localProcessHost{}, "cmd", "/c")
+
+	c := make(chan struct{})
+	cmd.RegisterAfterExitFun(func(_ context.Context) error {
+		close(c)
+		time.Sleep(50 * time.Millisecond)
+		return nil
+	})
+
+	err := cmd.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// call cmd.Wait to make sure after funs are called
+	done := make(chan error)
+	go func() {
+		done <- cmd.Wait()
+		close(done)
+	}()
+
+	err = cmd.Process.Wait()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
why are we doing both?

Reply:
So, that line, `cmd.Process.Wait`, will return once the process itself (`cmd /c`) returns, but `cmd.Wait` will return after the IO stuff is done (specifically the after exit callbacks).
So here, we wait for the process, then verify that the callback is called before the IO cleanup is finishe.
But the IO cleanup and all that is kicked off by `cmd.Wait`, so it has to be called to kick things off.